### ISSUE TITLE
DispNetCorr1D-K and DispNet-K missing (?) prototext files

### ISSUE THREAD

**Issue by OzzyW** [lmb-freiburg/dispnet-flownet-docker#4]
Hello again,

I have another question (I want to say the last...). In the paper it is said "We train DispNets on the FlyingThings3D dataset and then optionally fine-tune it on KITTI". I can not find the solver and training prototext files for the DispNetCorr1D-K and DispNet-K. 
1 ) Any clues on how to find these fine tuning prototexts or how was the fine tuning done? 
2) Also in the paper there are endpoint-error values (?) in parenthesis (in Table 3 in the KITTI train column), what do they mean? 
 
Thanks and Happy Holidays!

**Comment by nikolausmayer** [comment]
Hi again ;)

1) The `train.prototxt` are just the same as for the "non-K" nets. The only difference in finetuning is the crop size. As far as I can reconstruct (sorry, it's been a while), finetuning was done using a starting learning rate of 0.5, gamma=0.5, and steps at 200k, 350k, 500k until 620k. (But note that the cutoff was due to deadline, not because the net was totally converged)
2) The endpoint errors in parentheses are not really a fair comparison because that network was finetuned with KITTI 2015 training data, IIRC.

**Comment by OzzyW** [comment]
Hi again :) ,

1) Ok, cool! So fine-tuning is as easy as just changing the crop_width: 768 and crop_height: 384 accordingly in the DataAugmentation layer?  And a total of 620k steps for the KITTI +  1200K for the original FlyingThings, right? 

2) Do you recall how were the result with just training on the KITTI images without the FlyingThings data?

3) Also, what is the reason that not all layers in the deploy.prototxt are exactly the same (not only the names) as the train.prototxt, for example:
**DispNet deploy.tpl.prototext:**
...... 
layer {
  name: "Concat3"
  type: "Concat"
  bottom: "conv4_1"
  bottom: "deconv4"
  bottom: "upsampled_flow5_to_4"
  top: "blob31"
}
......
v.s.
**DispNet train.prototext:** 
... layer {
  name: "Concat3"
  type: "Concat"
  bottom: "conv5_1"
  bottom: "deconv5"
  bottom: "upsampled_flow6_to_5"
  top: "blob29"
}
....

**Comment by nikolausmayer** [comment]
1. To the best of my knowledge, yes. I don't remember our color augmentation settings for KITTI, though. It might make sense to restrict it as we know that the set is not very diverse.

2. Overfitting :) due to (at best) 400 samples and not a whole lot of augmentation. You'd most likely find that the top of the images gets no reasonable output (lack of groundtruth above a certain angle).

3. The `train.prototxt` has groundtruth inputs which are concatenated to some other blobs. That doesn't make much sense in deploy, so those layers are missing. The other layers' names are thus shifted.

**Comment by OzzyW** [comment]
Thanks so much for your replies!
About (2):  400 because it is not very diverse? (Even though there are many images available in the KITTI data set?) Would you be interested in the result?

**Comment by nikolausmayer** [comment]
KITTI has a lot of raw image data, but only ~200 training samples with groundtruth in each of the 2012 and 2015 releases. But yes, I'd be very interested :)

**Comment by nikolausmayer** [comment]
(closed due to inactivity; feel free to reply)

**Comment by OzzyW** [comment]
Hello again,

Sorry I didn't get to do this as I had some personal things to do and now I want to try this even if I get overfitting.

Can you point me on how to use a script on its own like the convert_imageset_and_disparity.cpp? Is there an easy way or just to see what libraries to add from caffe scripts without the makefile with trial and error.

Thanks in advance!

**Comment by nikolausmayer** [comment]
Hm, I'm not sure I understand the question. The necessary libraries for the whole codebase can be seen in the [Dockerfile](https://github.com/lmb-freiburg/dispnet-flownet-docker/blob/master/Dockerfile#L17).

**Comment by OzzyW** [comment]
I mean that I get a lot of errors when running the convert_imageset_and_disparity.cpp so is there an isolated script like the pfm to image converter maybe or if not what is the best way to convert external images to LMDB format so I can train them with this model?

**Comment by nikolausmayer** [comment]
Ah, I see. I'm sorry, but I don't know of any standalone converters. Even the simple imageset converter used in Caffe's MNIST example has these dependencies...
(we use our own in-house database, not LMDB)

**Comment by OzzyW** [comment]
OK, I figured it out (after some errors, mainly size mismatches and so on).. hope the results will be interesting and will update you. This git issue is a mess, should I open a new one?

**Comment by nikolausmayer** [comment]
That sounds like a good idea :) I'm closing this one then.
